Montage Start and End

tplatt77 opened the pull request:
Added Montage Start and Montage End fields to each of the montage import dialogs.
Allows user to only register and stitch a montage of the requested tiles in the range of montage start / end for column and row.

Commit:
Added montage start and end fields to Fiji import dialog

Signed-off-by: Tom Platt <tom.platt@bluequartz.net>

Commit:
Added montage start and end fields to Fiji import dialog

Signed-off-by: Tom Platt <tom.platt@bluequartz.net>

Commit:
Added montage start and end fields to generic import dialog

Signed-off-by: Tom Platt <tom.platt@bluequartz.net>

Commit:
Added montage start and end fields to generic import dialog

Signed-off-by: Tom Platt <tom.platt@bluequartz.net>

Commit:
Added montage start and end fields to Zeiss import dialog

Signed-off-by: Thomas Platt <tom.platt@bluequartz.net>

Commit:
Added montage start and end fields to Zeiss import dialog

Signed-off-by: Thomas Platt <tom.platt@bluequartz.net>

Commit:
Fixed an issue with ZeissListWidget not accessing Zeiss import API properly

Signed-off-by: Tom Platt <tom.platt@bluequartz.net>

Commit:
Merge branch 'topic/montageStartEnd' of https://github.com/tplatt77/SIMPLVtkLib into topic/montageStartEnd

Signed-off-by: Tom Platt <tom.platt@bluequartz.net>

# Conflicts:
#	SIMPLVtkLib/Dialogs/ImportFijiMontageDialog.cpp
#	SIMPLVtkLib/Dialogs/ImportFijiMontageDialog.h
#	SIMPLVtkLib/Dialogs/ImportZeissMontageDialog.cpp
#	SIMPLVtkLib/Dialogs/ImportZeissMontageDialog.h
#	SIMPLVtkLib/Dialogs/UI_Files/ImportFijiMontageDialog.ui
#	SIMPLVtkLib/Dialogs/UI_Files/ImportGenericMontageDialog.ui
#	SIMPLVtkLib/Dialogs/UI_Files/ImportZeissMontageDialog.ui
#	SIMPLVtkLib/Dialogs/ZeissListWidget.cpp
#	SIMPLVtkLib/QtWidgets/VSFilterFactory.cpp
#	SIMPLVtkLib/QtWidgets/VSFilterFactory.h

Commit:
Fixed names for the montage start and end for Fiji montage dialog

Signed-off-by: Tom Platt <tom.platt@bluequartz.net>

Commit:
Added an error label to the montage dialogs
* Shows an error message when the number of files is less than the montage size indicated by Montage Start and Montage End fields

Signed-off-by: Tom Platt <tom.platt@bluequartz.net>

Commit:
Added Montage Start and End fields to Zeiss Zen import dialog
* Added an error message to tell user when not enough files relative to montage start/end

Signed-off-by: Tom Platt <tom.platt@bluequartz.net>

joeykleingers reviewed (changes requested):
I tested importing the MNML_5 dataset using the Fiji importer.  Here were my settings:
* Montage start of (0, 0)
* Montage end of (1, 1)
* Data Display Type set to "Individual Tiles"

This configuration still loads the entire montage into the application.  This is because the montage start and end variables do not get taken into account until the registration step, which is not done when Outline Only or Individual Tiles is the data display type because that display type simply imports and displays all the tiles in their current positions.

To fix this issue, all the import filters need to be updated to take a montage start and end as well.

joeykleingers committed:
Update IMFViewer to use new montage start and end properties.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>

joeykleingers committed:
Update IMFViewer to use new montage start and end for stitching filter.

Uncomment PerformMontage registration filter code and create a prefix from a list of data containers so that it can be used to register the montage.

Signed-off-by: Joey Kleingers <joey.kleingers@bluequartz.net>
